spinalcordtoolbox/scripts/sct_detect_compression.py

```diff
@@ -0,0 +1,348 @@
+#!/usr/bin/env python
+#
+# Predict compression probability in a spinal cord MRI image using spinal cord shape metrics.
+# The script process axial slices at the level of intervertebral discs C3/C4 (value: 4) to C6/C7 (value: 7).
+# In other words, the script does *not* process all axial slices of the spinal cord segmentation.
+#
+# The script requires spinal cord segmentation (used to compute the shape metrics) and disc labels.
+#
+# More details in: https://pubmed.ncbi.nlm.nih.gov/35371944/
+#
+# Copyright (c) 2024 Polytechnique Montreal <www.neuro.polymtl.ca>
+# License: see the file LICENSE
+
+import os
+import sys
+import math
+from typing import Sequence
+import textwrap
+
+import numpy as np
+import pandas as pd
+
+from spinalcordtoolbox.process_seg import compute_shape
+from spinalcordtoolbox.aggregate_slicewise import aggregate_per_slice_or_level, func_wa, func_std, merge_dict
+from spinalcordtoolbox.centerline.core import ParamCenterline
+from spinalcordtoolbox.template import get_slices_from_vertebral_levels
+from spinalcordtoolbox.image import Image
+from spinalcordtoolbox.utils.fs import get_absolute_path, extract_fname
+from spinalcordtoolbox.utils.shell import SCTArgumentParser, Metavar
+from spinalcordtoolbox.utils.sys import init_sct, set_loglevel, printv
+
+
+# Currently, the model works only for discs C3/C4 (4) to C6/C7 (7)
+SUPPORTED_DISCS = {4: 'C3/C4', 5: 'C4/C5', 6: 'C5/C6', 7: 'C6/C7'}
+# Compression cut-offs --> compression probability
+#   >0.451 high probability of compression --> yes
+#   0.345-0.451 moderate probability of compression --> possible
+#   <0.345 low probability of compression --> no
+CUT_OFF_HIGH = 0.451
+CUT_OFF_MODERATE = 0.345
```

Comment:
Maybe we could expose these values in the argparse help and explain how they were determined? Something like:

```python
f"""
    Compression categories are determined based on the following cut-offs:
    
        - 'no'      : p <= {CUT_OFF_MODERATE}
        - 'possible': {CUT_OFF_MODERATE} < p <= {CUT_OFF_HIGH}
        - 'yes'     : p > {CUT_OFF_HIGH}

    These cut-off values were determined by [insert summary of methodology here]. More info in [REF].
"""
```

Right now, we are just outputting "yes"/"possible"/"no" in the CSV without much explanation of how those categories were determined.

Reply:
Good idea! 👍🏻 agree. done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4760/commits/36fc7da992d6e3330abff125586b349480a49c79